### PR TITLE
Add a new DataFeed named PaddleBoxDataFeed

### PR DESCRIPTION
**Pull request by ForFishes**
PaddleBoxDataFeed is a new datafeed. It has the following attributes：
1、It mainly has two functions to train
 - Merge_Pv_Instance(): It mainly merges instances to Pv(page view) and convey Pv to input_pv_channel_. 
 - Divide_Pv_Instance(): It mainly divides Pv(page view) to instances and convey instances to input_channel_. 

2、It mainly has two phases to train the dataset.
3、It can shift `enable_pv_predict` to select the mode.
4、It uses pointers to organize data.

### CODE REVIEW

File: paddle/fluid/framework/data_set.cc [hunk 1]
```diff
@@ -657,6 +774,92 @@ int DatasetImpl<T>::ReceiveFromClient(int msg_type, int client_id,
 // explicit instantiation
 template class DatasetImpl<Record>;
 
+void MultiSlotDataset::Divide_Pv_Instance() {
```
Comment (gavin1332) [inline paddle/fluid/framework/data_set.cc]: Invalid function name

File: paddle/fluid/framework/data_set.cc [hunk 2]
```diff
@@ -657,6 +774,92 @@ int DatasetImpl<T>::ReceiveFromClient(int msg_type, int client_id,
...
+    input_channel_->Close();
+  }
+  this->LocalShuffle();
+}
+
+void MultiSlotDataset::SetCurrentPhase(int current_phase) {
+  current_phase_ = current_phase;
+}
+
+void MultiSlotDataset::Merge_Pv_Instance() {
```
Comment (gavin1332) [inline paddle/fluid/framework/data_set.cc]: Invalid function name

File: paddle/fluid/framework/data_set.cc [hunk 3]
```diff
@@ -528,17 +618,30 @@ void DatasetImpl<T>::CreateReaders() {
     readers_[i]->SetFileList(filelist_);
     readers_[i]->SetParseInsId(parse_ins_id_);
     readers_[i]->SetParseContent(parse_content_);
+    readers_[i]->SetParseLogKey(parse_logkey_);
+    readers_[i]->SetEnablePvPredict(enable_pv_predict_);
+    // Notice: it is only valid for untest of test_paddlebox_datafeed.
+    // In fact, it does not affect the train process when paddle complied wit
```
Comment (hutuxian) [inline paddle/fluid/framework/data_set.cc]: typo?

File: paddle/fluid/framework/data_set.cc [hunk 4]
```diff
@@ -657,6 +774,92 @@ int DatasetImpl<T>::ReceiveFromClient(int msg_type, int client_id,
...
+
+void MultiSlotDataset::SetCurrentPhase(int current_phase) {
+  current_phase_ = current_phase;
+}
+
+void MultiSlotDataset::Merge_Pv_Instance() {
+  if (!input_channel_ || input_channel_->Size() == 0) {
+    return;
+  }
+  if (!enable_pv_predict_) {  // means to use Record
```
Comment (hutuxian) [inline paddle/fluid/framework/data_set.cc]: Should we move this outside of MergePV?

**Review by gavin1332 — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
lgtm for const_cast

**Pull request merged**